test/parseComment.js

```diff
@@ -294,11 +304,243 @@ describe('parseComment', function () {
+          textStyle: 'space',
+          start: 10,
+          end: 46
+        }
+      ],
+      source: [
+        {
+          number: 0,
+          source: '/** A link to {@link Something something awesome!}*/',
+          tokens: {
+            delimiter: '/**',
+            description: 'A link to {@link Something something awesome!}',
+            end: '*/',
+            lineEnd: '',
+            name: '',
+            postDelimiter: ' ',
+            postName: '',
+            postTag: '',
+            postType: '',
+            start: '',
+            tag: '',
+            type: ''
+          }
+        }
+      ],
+      problems: []
+    });
+  });
+
+  it('Handle inline type tag with "piped" text in description', function () {
+    const parsed = parseComment({
+      value: `* A link to {@link Something|something awesome!}`
+    });
+    expect(parsed).to.deep.equal({
+      description: 'A link to {@link Something|something awesome!}',
+      tags: [],
+      inlineTags: [
+        {
+          tag: 'link',
+          namepathOrURL: 'Something',
+          text: 'something awesome!',
+          textStyle: 'pipe',
+          start: 10,
+          end: 46
+        }
+      ],
+      source: [
+        {
+          number: 0,
+          source: '/** A link to {@link Something|something awesome!}*/',
+          tokens: {
+            delimiter: '/**',
+            description: 'A link to {@link Something|something awesome!}',
+            end: '*/',
+            lineEnd: '',
+            name: '',
+            postDelimiter: ' ',
+            postName: '',
+            postTag: '',
+            postType: '',
+            start: '',
+            tag: '',
+            type: ''
+          }
+        }
+      ],
+      problems: []
+    });
+  });
+
+  it('Handle inline type tag with "prefixed" text in description', function () {
+    const parsed = parseComment({
+      value: `* A link to [something awesome!]{@link Something}`
+    });
+    expect(parsed).to.deep.equal({
+      description: 'A link to [something awesome!]{@link Something}',
+      tags: [],
+      inlineTags: [
+        {
+          tag: 'link',
+          namepathOrURL: 'Something',
+          text: 'something awesome!',
+          textStyle: 'prefix',
+          start: 10,
+          end: 47
+        }
+      ],
+      source: [
+        {
+          number: 0,
+          source: '/** A link to [something awesome!]{@link Something}*/',
+          tokens: {
+            delimiter: '/**',
+            description: 'A link to [something awesome!]{@link Something}',
+            end: '*/',
+            lineEnd: '',
+            name: '',
+            postDelimiter: ' ',
+            postName: '',
+            postTag: '',
+            postType: '',
+            start: '',
+            tag: '',
+            type: ''
+          }
+        }
+      ],
+      problems: []
+    });
+  });
+
+  it('Handle plain inline type tag in tag', function () {
```

Comment:
To avoid further bloating the tests, I choose to implement just one of the text styles for the case of "inline tag in tag".

Reply:
Ok. I hope to add them later, as I'm big on full coverage even if we don't have it now (having been a port of code from `eslint-plugin-jsdoc`), but not needed for this PR.

Reply:
The file is reported to have 100% coverage with the tests I've added so far.